src/controller/BUILD.gn

```diff
@@ -25,26 +27,31 @@ static_library("controller") {
   output_name = "libChipController"
 
   sources = [
-    "AbstractDnssdDiscoveryController.cpp",
     "CHIPCluster.cpp",
     "CHIPCluster.h",
-    "CHIPCommissionableNodeController.cpp",
-    "CHIPCommissionableNodeController.h",
-    "CHIPDevice.cpp",
-    "CHIPDevice.h",
-    "CHIPDeviceController.cpp",
-    "CHIPDeviceController.h",
-    "CHIPDeviceControllerFactory.cpp",
-    "CHIPDeviceControllerFactory.h",
-    "DeviceAddressUpdateDelegate.h",
-    "DeviceDiscoveryDelegate.h",
-    "EmptyDataModelHandler.cpp",
-    "ExampleOperationalCredentialsIssuer.cpp",
-    "ExampleOperationalCredentialsIssuer.h",
-    "SetUpCodePairer.cpp",
-    "SetUpCodePairer.h",
   ]
 
+  if (chip_python_controller) {
+    sources += [
+      "AbstractDnssdDiscoveryController.cpp",
+      "CHIPCommissionableNodeController.cpp",
+      "CHIPCommissionableNodeController.h",
+      "CHIPDevice.cpp",
```

Comment:
This definitely looks odd.  Why do we not need CHIPDevice.cpp for _all_ controllers?

Reply:
same as above

Reply:
Done.